### ISSUE TITLE
Cannot get traces for block 21030627

### ISSUE THREAD

**Issue by artstar** [ledgerwatch/erigon#12432]
#### System information
Erigon version 2.60.8

#### Actual behaviour
Execute 
```
curl -X POST http://node.url/ -H 'Content-Type: application/json' -d '{"method":"trace_block","params":["0x140e6e3"],"id":1,"jsonrpc":"2.0"}'
```

Result:
```
{"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"first run for txIndex 85 error: insufficient funds for gas * price + value: address 0x0b92619DdE55C0cbf828d32993a7fB004E00c84B have 1414044108792614043 want 1414044806692614043"}}
```

#### Expected behaviour
Expected response is a list of traces without error.

**Comment by phyyou** [comment]
This issue seems to be related to #12092 reported earlier.

**Comment by MarcinZakrzewski13** [comment]
Are there any predictions as to when this problem will be fixed?

**Comment by zakirgl** [comment]
any updates here? or workarounds guys?

**Comment by jamesgithub1** [comment]
Hello. I am encountering the same issue. Is there any ETA for a fix? Does it affect all release versions?

**Comment by aliaksei-a** [comment]
try to use 
```
curl --location '<node_ip>:<node_port>' \
--header 'Content-Type: application/json' \
--data '{
	"jsonrpc":"2.0",
	"method":"debug_traceBlockByNumber",
	"params":[
		"0x140E6E3",
        { "tracer": "callTracer", "timeout": "300000ms" }
	],
	"id":"1"
}'
```
instead of `trace_block`
use erigon version 2.60.x, because for 2.59.x `debug_traceBlockByNumber` also returns error for tx `0x03d15392f4f774d0dbf8dff287991896cf39b171fb0b1da6c51d9045657a0cc9`:
`tracing failed: insufficient funds for gas * price + value:`
the point is that you need to handle traces in new way, so, you have to make remapping

**Comment by zeevo** [comment]
I am also experiencing this on version `2.60.6-d24e5d45`

**Comment by MarcinZakrzewski13** [comment]
> try to use
> 
> ```
> curl --location '<node_ip>:<node_port>' \
> --header 'Content-Type: application/json' \
> --data '{
> 	"jsonrpc":"2.0",
> 	"method":"debug_traceBlockByNumber",
> 	"params":[
> 		"0x140E6E3",
>         { "tracer": "callTracer", "timeout": "300000ms" }
> 	],
> 	"id":"1"
> }'
> ```
> 
> instead of `trace_block` use erigon version 2.60.x, because for 2.59.x `debug_traceBlockByNumber` also returns error for tx `0x03d15392f4f774d0dbf8dff287991896cf39b171fb0b1da6c51d9045657a0cc9`: `tracing failed: insufficient funds for gas * price + value:` the point is that you need to handle traces in new way, so, you have to make remapping

Or `trace_transaction` if you can skip individual transactions and are dependent on the data model.

**Comment by nate-staked** [comment]
also hitting this, would love a fix, happy to test

**Comment by racket2000** [comment]
+1

I am encountering this exact issue as well, and would greatly appreciate a fix or workaround.

**Comment by tjayrush** [comment]
> the point is that you need to handle traces in new way, so, you have to make remapping

I'm curious about this. First, are you one of the devs on Erigon?

In many cases, "handling traces in a new way" is not possible. 

Perhaps I'm just misunderstanding what you're saying. Sorry if so.

**Comment by aliaksei-a** [comment]
> > the point is that you need to handle traces in new way, so, you have to make remapping
> 
> I'm curious about this. First, are you one of the devs on Erigon?
> 
> In many cases, "handling traces in a new way" is not possible.
> 
> Perhaps I'm just misunderstanding what you're saying. Sorry if so.

I am not an Erigon dev. I've just faced an issue as others in this thread. I wanted to get traces for block 21030627 but
`trace_block` threw and error and I suggested to use `debug_traceBlockByNumber` instead because looks like all data from `trace_block` can be fetched from `debug_traceBlockByNumber` (at least in my case), but it is structured in another way.

**Comment by racytech** [comment]
Hi everyone, this issue is being investigated. Since we are moving towards erigon3 (which has more optimized state history and faster sync time then version 2.x.x), we made many changes and some parts of the code in versions 2.x.x may left without necessary attention. As soon as the problem is detected it will be fixed. Thank you for your patience

**Comment by tjayrush** [comment]
Thanks for letting us know. I've been meaning to upgrade, so this will be impetus. Question for you though, first: I'm certain moving to 3.x requires a full re-sync, but does anyone have any idea how long that takes for a full archive node? Just need to plan ahead.

**Comment by AskAlexSharov** [comment]
> Thanks for letting us know. I've been meaning to upgrade, so this will be impetus. Question for you though, first: I'm certain moving to 3.x requires a full re-sync, but does anyone have any idea how long that takes for a full archive node? Just need to plan ahead.

Erigon3 does downloading most of the data. So speed depends on your network speed and —torrent.download.rate, we did manage sync ethmInnet in 8hours (and polygon mainnet in 2 days): https://erigon.substack.com/p/erigon-3-alpha-2-introducing-blazingly

**Comment by yperbasis** [comment]
Fixed by PR #12559 and will be released in v2.60.10.
